modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java

```diff
@@ -0,0 +1,546 @@
+        }
+
+        if (!migrationNeeded) {
+            log.warn("Database was already migrated. Consider to disable migration in application settings.");
+
+            return;
+        }
+
+        log.info("Migration started...");
+
+        MongoClient mongoClient = MongoClients.create();
+
+        try {
+            mongoDb = mongoClient.getDatabase(mongoDbName);
+
+            migrateAccounts();
+
+            migrateEx();
+
+            log.info("Migration finished!");
+        }
+        catch (Throwable e) {
+            log.error("Migration failed", e);
+        }
+        finally {
+            U.closeQuiet(mongoClient);
+        }
+    }
+
+    /**
+     * Extension point for migration of additional objects.
+     */
+    protected void migrateEx() {
+        // No-op.
+    }
+
+    /**
+     * @param doc Mongo document.
+     * @param accId Account ID.
+     * @return Account to save.
+     */
+    protected Account createAccount(Document doc, UUID accId) {
+        Account acc = new Account();
+
+        acc.setId(accId);
+        acc.setEmail(doc.getString("email"));
+        acc.setPassword("PBKDF2:" + doc.getString("salt") + ":" + doc.getString("hash"));
+        acc.setFirstName(doc.getString("firstName"));
+        acc.setLastName(doc.getString("lastName"));
+        acc.setPhone(doc.getString("phone"));
+        acc.setCompany(doc.getString("company"));
+        acc.setCountry(doc.getString("country"));
+        acc.setToken(doc.getString("token"));
+        acc.setResetPasswordToken(doc.getString("resetPasswordToken"));
+
+        acc.setAdmin(doc.getBoolean("admin", false));
+
+        return acc;
+    }
+
+    /**
+     * Migrate accounts.
+     */
+    private void migrateAccounts() {
+        MongoCollection<Document> accountsCollection = mongoDb.getCollection("accounts");
+        MongoCollection<Document> spacesCollection = mongoDb.getCollection("spaces");
+
+        log.info(off(1, "Accounts to migrate: " + accountsCollection.countDocuments()));
+
+        try (MongoCursor<Document> cursor = accountsCollection.find().iterator()) {
+            while (cursor.hasNext()) {
+                Document accMongo = cursor.next();
+
+                ObjectId mongoAccId = accMongo.getObjectId("_id");
+                String email = accMongo.getString("email");
+
+                Document space = spacesCollection.find(
+                    Filters.and(
+                        Filters.eq("owner", mongoAccId),
+                        Filters.eq("demo", false)
+                    )
+                ).first();
+
+                if (space == null) {
+                    log.warn("Space not found [owner=" + mongoAccId + ", email=" + email + "]");
+
+                    continue;
+                }
+
+                log.info(off(2, "Migrating account [_id=" + mongoAccId + ", email=" + email + "]"));
+
+                UUID accId = UUID.randomUUID();
+
+                Account acc = createAccount(accMongo, accId);
```

Comment:
Move this logic to `migrateAccount` and remove `migrateAccountEx`

Reply:
I don't think it is a good idea, I need to override this method for "GG plugin" migration.

Reply:
Fixed.